demos/http/http_demo_basic_tls/http_demo_basic_tls.c

```diff
@@ -0,0 +1,810 @@
+    }
+
+    if( cwd != NULL )
+    {
+        free( cwd );
+    }
+
+    if( pRootCaFile != NULL )
+    {
+        ( void ) fclose( pRootCaFile );
+    }
+
+    if( pRootCa != NULL )
+    {
+        X509_free( pRootCa );
+    }
+
+    if( pSslSetup != NULL )
+    {
+        SSL_CTX_free( pSslSetup );
+    }
+
+    /* Log failure or success and return the correct exit status. */
+    if( sslStatus == 0 )
+    {
+        LogError( ( "Failed to establish a TLS connection: Host=%.*s.",
+                    ( int32_t ) SERVER_HOST_LENGTH,
+                    SERVER_HOST ) );
+        return EXIT_FAILURE;
+    }
+    else
+    {
+        LogInfo( ( "Established a TLS connection: Host=%.*s.\n\n",
+                   ( int32_t ) SERVER_HOST_LENGTH,
+                   SERVER_HOST ) );
+        return EXIT_SUCCESS;
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t transportSend( HTTPNetworkContext_t * pNetworkContext,
+                              const void * pBuffer,
+                              size_t bytesToSend )
+{
+    int32_t bytesSent = 0;
+    int pollStatus = 0;
+    struct pollfd fileDescriptor =
+    {
+        .events  = POLLOUT,
+        .revents = 0
+    };
+
+    /* Set the file descriptor for poll. */
+    fileDescriptor.fd = SSL_get_fd( pNetworkContext->pSslContext );
+
+    /* Poll the file descriptor to check if SSL_Write can be done now. */
+    pollStatus = poll( &fileDescriptor, 1, TRANSPORT_SEND_RECV_TIMEOUT_MS );
+
+    if( pollStatus > 0 )
+    {
+        bytesSent = ( int32_t ) SSL_write( pNetworkContext->pSslContext, pBuffer, bytesToSend );
+    }
+    else if( pollStatus == 0 )
+    {
+        LogDebug( ( "Timed out while polling SSL socket for write buffer availability." ) );
+    }
+    else
+    {
+        LogError( ( "Polling of the SSL socket for write buffer availability failed:"
+                    " status=%d",
+                    pollStatus ) );
+        bytesSent = -1;
+    }
+
+    return bytesSent;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t transportRecv( HTTPNetworkContext_t * pNetworkContext,
+                              void * pBuffer,
+                              size_t bytesToRecv )
+{
+    int32_t bytesReceived = 0;
+    int pollStatus = -1, bytesAvailableToRead = 0;
+    struct pollfd fileDescriptor =
+    {
+        .events  = POLLIN | POLLPRI,
+        .revents = 0
+    };
+
+    /* Set the file descriptor for poll. */
+    fileDescriptor.fd = SSL_get_fd( pNetworkContext->pSslContext );
+
+    /* Check if there are any pending data available for read. */
+    bytesAvailableToRead = SSL_pending( pNetworkContext->pSslContext );
+
+    /* Poll only if there is no data available yet to read. */
+    if( bytesAvailableToRead <= 0 )
+    {
+        pollStatus = poll( &fileDescriptor, 1, TRANSPORT_SEND_RECV_TIMEOUT_MS );
+    }
+
+    /* SSL read of data. */
```

Comment:
Would you please enhance the comment to say:

```
/* bytesAvailableToRead > 0 means that there was pending data to be read.
 * pollStatus > 0 means that there was no pending data but it became available 
 * during polling. Read the available data. */
```

In line of the above, I think the order of the check should be changed so that the short-circuiting only checks for pollStatus, if the poll was called:

```c
if( ( bytesAvailableToRead > 0 ) || ( pollStatus > 0 ) )
```